vm/context.ts

```diff
@@ -399,6 +407,21 @@ module J2ME {
       return frames[frames.length - 1];
     }
 
+    popFrame(): Frame {
+      var frame = this.frames.pop();
+      if (profile) {
+        this.leaveMethodTimeline(frame.methodInfo.implKey, MethodType.Interpreted);
+      }
+      return frame;
+    }
+
+    pushFrame(frame: Frame) {
+      if (profile) {
+        this.enterMethodTimeline(frame.methodInfo.implKey, MethodType.Interpreted);
+      }
+      this.frames.push(frame);
+    }
```

Comment:
I didn't notice anything on a quick desktop tests. I'll apply your comments and run on device in a bit.